ingeniamotion/communication.py

```diff
@@ -453,15 +465,68 @@ def get_ifname_by_index(self, index: int) -> str:
         """
         return self.__get_adapter_name(index)
 
-    @staticmethod
-    def get_interface_name_list() -> List[str]:
+    @classmethod
+    def get_interface_name_list(cls) -> List[str]:
         """Get interface list.
 
         Returns:
             List with interface readable names.
 
         """
-        return [x.nice_name for x in ifaddr.get_adapters()]
+        network_adapters = cls.get_network_adapters()
+        return list(network_adapters)
+
+    @staticmethod
+    def get_network_adapters() -> Dict[str, str]:
+        """Get the detected network adapters.
+
+        Returns:
+            Dictionary with interface readable names as keys and GUIDs as values.
+
+        """
+        network_adapters = []
+        for adapter in ifaddr.get_adapters():
+            if isinstance(adapter.name, bytes):
+                adapter_guid = bytes.decode(adapter.name)
+            else:
+                adapter_guid = adapter.name
+            network_adapters.append(NetworkAdapter(adapter.index, adapter.nice_name, adapter_guid))
+        if RUNNING_ON_WINDOWS:
+            # When using WMI within threads it is required to initialize the COM objects
+            # https://stackoverflow.com/a/14428972
+            # The order of imports is also important
+            # https://stackoverflow.com/a/46606527
+            import pythoncom
+
+            pythoncom.CoInitialize()
+            import wmi
+
+            for adapter in [
+                NetworkAdapter(o.interfaceindex, o.Name, o.GUID)
+                for o in wmi.WMI().query(
+                    "select Name, guid, interfaceindex from Win32_NetworkAdapter"
+                )
+                if o.GUID is not None
+            ]:
+                if adapter not in network_adapters:
+                    network_adapters.append(adapter)
+
+            interface_name_counter = Counter(
+                [adapter.interface_name for adapter in network_adapters]
+            )
+
+            if interface_name_counter.most_common(1)[0][1] > 1:
```

Comment:
What are those keys?

Reply:
- most_common(1): This method returns a list of the n most common elements and their counts from the most common to the least. In this case, 1 means it returns a list with the single most common element.

- [0]: This accesses the first element of the list returned by most_common(1). Since most_common(1) returns a list with one tuple, [0] gets that tuple.

- [1]: This accesses the second element of the tuple, which is the count of the most common element.